aws/aws.go

```diff
@@ -178,29 +178,42 @@ func NewDefaultV2Config(ctx context.Context) (awsv2.Config, error) {
 //   - profile: The shared config profile to use; sets SharedConfigProfile.
 //   - endpoint: The AWS service endpoint to send HTTP request.
 func V2ConfigFromURLParams(ctx context.Context, q url.Values) (awsv2.Config, error) {
+	var endpoint string
+	var hostnameImmutable bool
 	var opts []func(*awsv2cfg.LoadOptions) error
 	for param, values := range q {
 		value := values[0]
 		switch param {
+		case "awsHostnameImmutable":
```

Comment:
Let's use "hostname_immutable", there's no need for "aws" prefix and URL params aren't case sensitive.